catalyst/loggers/neptune.py

```diff
@@ -0,0 +1,256 @@
+    conflict_keys = []
+    processed_metrics = dict(metrics)
+    for k in processed_metrics.keys():
+        if k.endswith("/std"):
+            k_stripped = k[:-4]
+            k_val = k_stripped + "/val"
+            if k_val not in processed_metrics.keys():
+                processed_metrics[k_val] = processed_metrics.pop(k_stripped)
+    for k in processed_metrics.keys():
+        for j in processed_metrics.keys():
+            if j.startswith(k) and j != k and k not in conflict_keys:
+                conflict_keys.append(k)
+    for i in conflict_keys:
+        processed_metrics[i + "_val"] = processed_metrics.pop(i)
+    return processed_metrics
+
+
+class NeptuneLogger(ILogger):
+    """
+    NeptuneLogger docs
+    """
+
+    def __init__(
+        self, base_namespace=None, api_token=None, project=None, run=None, **neptune_run_kwargs
+    ):
+        if base_namespace is None:
+            self.base_namespace = "experiment"
+        else:
+            self.base_namespace = base_namespace
+        self._api_token = api_token
+        self._project = project
+        self._neptune_run_kwargs = neptune_run_kwargs
+        if run is None:
+            self.run = neptune.init(
+                project=self._project, api_token=self._api_token, **self._neptune_run_kwargs
+            )
+        else:
+            self.run = run
+
+    def _log_metrics(self, metrics: Dict[str, float], neptune_path: str, step: int):
+        for key, value in metrics.items():
+            self.run[f"{neptune_path}/{key}"].log(value=float(value), step=step)
+
+    def _log_image(self, image: np.ndarray, neptune_path: str):
+        self.run[neptune_path].log(neptune.types.File.as_image(image))
+
+    def _log_artifact(self, artifact: object, path_to_artifact: str, neptune_path: str):
+        if artifact is not None:
+            self.run[neptune_path].upload(neptune.types.File.as_pickle(artifact))
+        elif path_to_artifact is not None:
+            self.run[neptune_path].upload(path_to_artifact)
+
+    def log_metrics(
+        self,
+        metrics: Dict[str, float],
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        global_epoch_step: int = 0,
+        global_batch_step: int = 0,
+        global_sample_step: int = 0,
+        # stage info
+        stage_key: str = None,
+        stage_epoch_len: int = 0,
+        stage_epoch_step: int = 0,
+        stage_batch_step: int = 0,
+        stage_sample_step: int = 0,
+        # loader info
+        loader_key: str = None,
+        loader_batch_len: int = 0,
+        loader_sample_len: int = 0,
+        loader_batch_step: int = 0,
+        loader_sample_step: int = 0,
+    ) -> None:
+        """Logs batch, epoch and loader metrics to Neptune."""
+        if scope == "batch":
+            neptune_path = "/".join([self.base_namespace, stage_key, loader_key, scope])
+            self._log_metrics(metrics=metrics, neptune_path=neptune_path, step=global_batch_step)
```

Comment:
```suggestion
            self._log_metrics(metrics=metrics, neptune_path=neptune_path, step=global_sample_step)
```

per-sample counter allows you to compare runs with different batch sizes :)